frame/support/src/traits/hooks.rs

```diff
@@ -82,11 +82,17 @@ pub trait OnIdle<BlockNumber> {
 #[impl_for_tuples(30)]
 impl<BlockNumber: Clone> OnIdle<BlockNumber> for Tuple {
 	fn on_idle(n: BlockNumber, remaining_weight: crate::weights::Weight) -> crate::weights::Weight {
-		let mut weight = 0;
+		let mut on_idle_functions = sp_std::vec::Vec::<fn(BlockNumber, crate::weights::Weight) -> crate::weights::Weight>::new();
 		for_tuples!( #(
-			let adjusted_remaining_weight = remaining_weight.saturating_sub(weight);
-			weight = weight.saturating_add(Tuple::on_idle(n.clone(),  adjusted_remaining_weight));
+			on_idle_functions.push(Tuple::on_idle);
 		)* );
+		let mut weight = 0;
+		for pallet_index in 0..on_idle_functions.len() {
+			let adjusted_remaining_weight = remaining_weight.saturating_sub(weight);
+			// let index = (pallet_index + n) % on_idle_functions.len();
+			let on_idle = on_idle_functions[pallet_index];
+			weight = weight.saturating_add(on_idle(n.clone(), adjusted_remaining_weight));
```

Comment:
I'm surprised the `clone()` is necessary, are you sure?

Reply:
Yes, because it is inside the loop so first element will be borrowed, so for the next iteration we have a compilation error
```
  84 |     fn on_idle(n: BlockNumber, remaining_weight: crate::weights::Weight) -> crate::weights::Weight {
     |                - move occurs because `n` has type `BlockNumber`, which does not implement the `Copy` trait
  ...
  94 |             let n1 = n.clone() % BlockNumber::from(on_idle_functions.len() as u32);
     |                      ^ value borrowed here after move
  ...
  99 |             weight = weight.saturating_add(on_idle(n, adjusted_remaining_weight));
     |                                                    - value moved here, in previous iteration of loop
     ```

Reply:
IMO it would be fine to require `BlockNumber` to be `Copy` here, but ok :shrug: